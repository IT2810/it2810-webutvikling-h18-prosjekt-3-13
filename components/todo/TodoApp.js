import React from "react";
import { Text, View } from "react-native";
import { List } from "react-native-elements";

import TodoModel from "./TodoModel";
import PartList from "./PartList";
import InputBox from "./InputBox";

import { styles } from "../../screens/styles.js";

class TodoApp extends React.Component {
  constructor(props) {
    super(props);

    let initList = [];

    if (this.props.test) {
      for (i = 0; i < 4; i++) {
        initList.push(new TodoModel("Heihei", false));
      }
    }

    this.state = {
      dataList: initList
    };
  }

  addTodoItem = task => {
    const midList = this.state.dataList.slice();
    midList.push(new TodoModel(task, false));
    this.setState({
      dataList: midList
    });
  };

  removeTodoItem = task => {
    const midList = this.state.dataList.slice();
    const taskIndex = midList.indexOf(task);
    midList.splice(taskIndex, 1);
    this.setState({
      dataList: midList
    });
  };

  changeCompleted = task => {
    const midList = this.state.dataList.slice();
    const taskIndex = midList.indexOf(task);
    let midTask = task;
    midTask.completed = !midTask.completed;
    midList.splice(taskIndex, 1, midTask);
    this.setState({
      dataList: midList
    });
  };

  handleDelete = task => {
    this.removeTodoItem(task);
  };

  handleChange = task => {
    this.changeCompleted(task);
  };

  handleAdd = task => {
    this.addTodoItem(task);
  };

  render() {
    console.log("UUUUUU:", this.props.screenProps);
    const completed = this.state.dataList.filter(task => task.completed);
    const notCompleted = this.state.dataList.filter(task => !task.completed);

    renderLists = () => {
      if (this.state.dataList.length) {
        return (
          <View style={styles(this.props.screenProps).toDoLists1}>
            <PartList
              screenProps={this.props.screenProps}
              list={notCompleted}
              handleDelete={this.handleDelete}
              handleChange={this.handleChange}
            />
            <PartList
              screenProps={this.props.screenProps}
              list={completed}
              handleDelete={this.handleDelete}
              handleChange={this.handleChange}
            />
          </View>
        );
      } else {
        return (
          <View style={styles(this.props.screenProps).toDoLists2}>
            <Text
              style={{ fontSize: 50, color: "lightgrey", textAlign: "center" }}
            >
              All todo's completed
            </Text>
          </View>
        );
      }
    };
    return (
      <View style={styles(this.props.screenProps).toDoLists3}>
        <InputBox
          screenProps={this.props.screenProps}
          style={{
            flex: 1
          }}
          onAdd={this.handleAdd}
        />
        {renderLists()}
      </View>
    );
  }
}

module.exports = TodoApp;
