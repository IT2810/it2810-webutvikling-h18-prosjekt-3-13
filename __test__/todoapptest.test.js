import React from "react";
import TodoApp from "../components/todo/TodoApp";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <TodoApp
        test={true}
        screenProps={{
          someText: "some text",
          handleThemeChange: this.handleThemeChange,
          bg: "white",
          color: "black"
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
