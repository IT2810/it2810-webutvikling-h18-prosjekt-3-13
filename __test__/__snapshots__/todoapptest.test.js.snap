// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
      "paddingTop": 70,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      blurOnSubmit={false}
      onChange={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Add a task"
      placeholderTextColor="black"
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#eee",
          "borderRadius": 8,
          "borderWidth": 1,
          "color": "black",
          "flex": 1,
          "fontSize": 16,
          "height": 36,
          "margin": 8,
          "paddingLeft": 10,
        }
      }
      value=""
    />
    <View
      style={
        Object {
          "display": "none",
        }
      }
    >
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#00aced",
                  "fontSize": 35,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 7,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "blue",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
        >
          <View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#00aced",
                      "fontSize": 35,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
          <View
            width={16}
          />
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "paddingLeft": "2%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "black",
                  }
                }
              >
                Heihei
              </Text>
            </View>
          </View>
          <View
            width={16}
          />
          <View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#00aced",
                      "fontSize": 26,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
        >
          <View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#00aced",
                      "fontSize": 35,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
          <View
            width={16}
          />
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "paddingLeft": "2%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "black",
                  }
                }
              >
                Heihei
              </Text>
            </View>
          </View>
          <View
            width={16}
          />
          <View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#00aced",
                      "fontSize": 26,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
        >
          <View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#00aced",
                      "fontSize": 35,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
          <View
            width={16}
          />
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "paddingLeft": "2%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "black",
                  }
                }
              >
                Heihei
              </Text>
            </View>
          </View>
          <View
            width={16}
          />
          <View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#00aced",
                      "fontSize": 26,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
        >
          <View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#00aced",
                      "fontSize": 35,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
          <View
            width={16}
          />
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "paddingLeft": "2%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "black",
                  }
                }
              >
                Heihei
              </Text>
            </View>
          </View>
          <View
            width={16}
          />
          <View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#00aced",
                      "fontSize": 26,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "blue",
        }
      }
    />
  </View>
</View>
`;
